processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMapping.java

```diff
@@ -61,12 +63,25 @@ public static BeanMapping fromPrism(BeanMappingPrism beanMapping, ExecutableElem
             beanMapping.qualifiedByName(),
             resultTypeIsDefined ? beanMapping.resultType() : null );
 
-        return new BeanMapping(cmp, nullValueMappingStrategy );
+        //TODO Do we want to add the reporting policy to the BeanMapping as well? To give more granular support?
+        return new BeanMapping( cmp, nullValueMappingStrategy, null );
     }
 
-    private BeanMapping( SelectionParameters selectionParameters, NullValueMappingStrategyPrism nvms ) {
+    /**
+     * This method should be used to generate BeanMappings for our internal generated Mappings. Like forged update
```

Comment:
Basically all our forged mappings. However, after thinking a bit more. We might _only_ need to have `ReportingPolicy.IGNORE` for composite methods.